docs/modules/data-structures/pages/vector-collections.adoc

```diff
@@ -0,0 +1,495 @@
+However, for a multi-index vector collection, specifying the name of the index to search in is necessary.
+
+Example for single-index vector collection:
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+CompletionStage<SearchResults<String, String>> results = vectorCollection.searchAsync(
+        VectorValues.of(new float[] {0f, 0f, 0.2f, -0.3f, 1.2f}),
+        SearchOptions.builder()
+            .limit(5)
+            .includeVectors()
+            .includeValue()
+            .build()
+);
+----
+--
+====
+
+Example for multi-index vector collection:
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+CompletionStage<SearchResults<String, String>> results = vectorCollection.searchAsync(
+        VectorValues.of("glove-index", new float[] {0f, 0f, 0.2f, -0.3f, 1.2f}),
+        SearchOptions.builder()
+            .limit(5)
+            .includeVectors()
+            .includeValue()
+            .build()
+);
+----
+--
+====
+
+=== Similarity search options
+Search parameters are passed as an argument `searchOptions` to the searchAsync method.
+
+.Search options
+[cols="1,2,1",options="header"]
+|===
+|Option|Description|Default
+
+|limit
+|How many results should be returned in search result
+|`1`
+
+|includeValue
+|Include the user value in the search result.
+|`FALSE`
+
+
+|includeVectors
+|Include the vectors values in the search result.
+|`FALSE`
+
+|hints
+|Extra hints for the search.
+|`NULL`
+
+|===
+
+
+.Available hints
+[cols="1,2",options="header"]
+|===
+|Hint|Description
+
+|partitionLimit
+|Number of results to fetch from partition.
+
+|memberLimit
+|Number of results to fetch from member in 2-stage search.
+
+|singleStage
+|Force use of 1-stage search.
+
+|===
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+var options = SearchOptions.builder()
+                .limit(10)
+                .includeValue()
+                .includeVectors()
+                .hint("partitionLimit", 1)
+                .build();
+----
+--
+====
+
+WARNING: Hints allow fine-tuning some aspects of search execution but are subject to change and may be removed in future versions.
+
+== Manage collection
+
+=== Optimize collection
+
+An optimization operation may be needed in the following cases:
+
+* To permanently delete vectors that were marked for removal.
+* After adding a significant number of vectors.
+* The collection returns fewer vectors than expected.
+
+WARNING: The optimization operation can be a time-consuming and resource-intensive process, and no mutating operations are allowed during this process.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+CompletionStage<Void> result = vectorCollection.optimizeAsync("glove-index");
+----
+--
+====
+
+=== Clear collection
+To remove all vectors and values from the vector collection use the method `clearAsync()`.
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+CompletionStage<Void> result = vectorCollection.clearAsync();
+----
+--
+====
+
```

Comment:
```suggestion

== Limitations in beta version

In current version Vector Collection is provided as Beta feature with some limitations, the most significant being:

1. API may be changed in future versions
2. Lack of rolling-upgrade compatibility guarantees for vector collections. It may be required to delete existing vector collections before migrating to future version of Hazelcast
3. Lack of fault tolerance, backups cannot yet be configured. However, data in collections is migrated to other cluster members in case of graceful shutdown and new member joining the cluster, so in case of lack of failures normal cluster maintenance (e.g. rolling restart) is possible without data loss.
4. Only on-heap storage of vector collections is available

```

Point 3 depends on https://github.com/hazelcast/hazelcast-mono/pull/2218